src/btf.cpp

```diff
@@ -684,7 +684,7 @@ std::map<std::string, std::vector<std::string>> BTF::get_params_from_btf(
       params[func_name].push_back(type + " " + arg_name);
     }
 
-    if (!t->type)
+    if (!t->type || is_raw_tracepoint)
```

Comment:
As an aside, the `is_raw_tracepoint` parameter is a flag for me. You should just submit and get the bug fixed, but in terms of API design, this doesn't make sense. It's undocumented and leaking unexpected semantics across the object boundary. If possible, I would try to encode this in some way that makes sense within the class API, not that is meaningful to the caller, e.g. `omit_retval` or something. (Otherwise the reader of this function needs to work way, way harder to understand if the behavior is correct, if they are able to at all.)

(I realize that it's also used above, but you could provide separate parameters, e.g. one could also provide an optional function that maps from the BTF function name to `func_name`, although I'm not sure why you wouldn't do this from the caller?)

Reply:
@amscanne You mean **red** flag 😸  - but yes, I agree - I think I had it this way originally and I changed it back for some reason - let me fix that in a follow-up PR.